gprofiler/merge.py

```diff
@@ -96,33 +185,35 @@ def parse_perf_script(script: str):
             match = SAMPLE_REGEX.match(sample)
             if match is None:
                 raise Exception("Failed to match sample")
-            yield match.groupdict()
+            sample_dict = match.groupdict()
+
+            pid = int(sample_dict["pid"])
+            comm = sample_dict["comm"]
+            stack = sample_dict["stack"]
+            if stack is not None:
+                pid_to_collapsed_stacks_counters[pid][collapse_stack(comm, stack)] += 1
+            pid_to_comm.setdefault(pid, comm)
```

Comment:
Why `setdefault`?

Reply:
Pretty sure I already answered a similar one - it's to set the comm only if the PID doesn't exist yet in the `pid_to_comm` dict in a single atomic operation

Reply:
Hmm haven't seen that previous one then. Okay

Reply:
I'm not sure I understand tho. You never actually write to this dict? Only with `setdefault` and with `update`

Reply:
The `setdefault` does the writing. I could replace this code with:
```python
if pid not in pid_to_comm:
    pid_to_comm[pid] = comm
```

Reply:
Yes, but you only write to it in this function. I thought `setdefault` overrides, but apparently it doesn't:
```
In [1]: x = {}                                                                                                                                                                                             

In [2]: x.setdefault("y", 1)                                                                                                                                                                               
Out[2]: 1

In [3]: x["y"]                                                                                                                                                                                             
Out[3]: 1

In [4]: x.setdefault("y", 2)                                                                                                                                                                               
Out[4]: 1

In [5]: x["y"]                                                                                                                                                                                             
Out[5]: 1
```

Alright then. Previous code used to take the last `comm` and now you're taking the first `comm`. What's the reason?

Reply:
It's arbitrary either way.

Reply:
Hmm. Yup. Well, we will need to improve it somehow. It's problematic

Reply:
Yields weird stacks